Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift

```diff
@@ -0,0 +1,136 @@
+/* Copyright 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import SwiftUI
+import DesignSystem
+import struct Shared.AppConstants
+import LocalAuthentication
+
+struct BiometricView: View {
+  var keyringStore: KeyringStore
+  var completion: () -> Void
+  
+  @State private var biometricError: OSStatus?
+  
+  private var biometricsIcon: Image? {
+    let context = LAContext()
+    if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil) {
+      switch context.biometryType {
+      case .faceID:
+        return Image(braveSystemName: "leo.face.id")
+      case .touchID:
+        return Image(braveSystemName: "leo.biometric.login")
+      case .none:
+        return nil
+      @unknown default:
+        return nil
+      }
+    }
+    return nil
+  }
+  
+  private var biometricName: String {
+    let context = LAContext()
+    if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil) {
+      switch context.biometryType {
+      case .faceID:
+        return Strings.Wallet.biometricsSetupFaceId
+      case .touchID:
+        return Strings.Wallet.biometricsSetupTouchId
+      default:
+        return ""
+      }
+    }
+    return ""
+  }
+
+  var body: some View {
+    VStack {
+      ZStack {
+        Circle()
+          .strokeBorder(Color(.braveInfoBorder).opacity(0.3))
+          .background(
+            Circle()
+              .foregroundColor(Color(.braveInfoBackground).opacity(0.5))
+          )
+          .frame(width: 240, height: 240)
+        Rectangle()
+          .frame(width: 96, height: 96)
+          .foregroundColor(Color(.braveBackground))
+          .cornerRadius(20)
+        if let biometricsIcon {
+          biometricsIcon
+            .resizable()
+            .aspectRatio(contentMode: .fit)
+            .frame(width: 52, height: 52)
+            .foregroundColor(Color(.braveInfoLabel))
+        }
+      }
+      Group {
+        Text(String.localizedStringWithFormat(Strings.Wallet.biometricsSetupTitle, biometricName))
+          .font(.title)
+          .foregroundColor(.primary)
+          .padding(.bottom, 10)
+        Text(String.localizedStringWithFormat(Strings.Wallet.biometricsSetupSubTitle, biometricName))
+          .font(.body)
+          .foregroundColor(Color(.secondaryBraveLabel))
+      }
+      .fixedSize(horizontal: false, vertical: true)
+      .multilineTextAlignment(.center)
+      .padding(.top, 28)
+      Button {
+        // Store password in keychain
+        if let password = keyringStore.passwordToSaveInBiometric, case let status = keyringStore.storePasswordInKeychain(password),
```

Comment:
just double checking for my sanity - is the keystore in the OS keychain sandboxed per app?

Reply:
an application can access only its own keychain items as it depends on app's bundle id and the id is unique.